java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -311,20 +318,27 @@ public String linkUrl() {
   }
 
   private void addMethodPropertiesToConfigMethodProperties(
-      Set<Map<String, Object>> methodProperties) {
+      Set<Map<String, Object>> methodProperties, boolean isArgumentIndexOptional)
+      throws ParseException {
     for (Map<String, Object> methodProperty : methodProperties) {
-      String flagType = getValueStringFromMap(methodProperty, FLAG_TYPE_KEY);
       String methodName = getValueStringFromMap(methodProperty, METHOD_NAME_KEY);
-      if (flagType != null && methodName != null) {
+      if (methodName != null && getValueStringFromMap(methodProperty, FLAG_TYPE_KEY) != null) {
+        if (!isArgumentIndexOptional && getArgumentIndexFromMap(methodProperty) == null) {
+          throw new ParseException(
+              "methodProperty did not have argumentIndex. Use Piranha:ArgumentIndexOptional if required. Check:"
+                  + methodProperty,
+              0);
+        }
         List<Map<String, Object>> methodPropertiesForMethodName =
             configMethodProperties.get(methodName);
-        if (methodPropertiesForMethodName != null) {
-          methodPropertiesForMethodName.add(methodProperty);
-        } else {
-          List<Map<String, Object>> methodPropertiesForNewMethodName = new ArrayList<>();
-          methodPropertiesForNewMethodName.add(methodProperty);
-          configMethodProperties.put(methodName, methodPropertiesForNewMethodName);
+        if (methodPropertiesForMethodName == null) {
+          methodPropertiesForMethodName = new ArrayList<>();
+          configMethodProperties.put(methodName, methodPropertiesForMethodName);
         }
+        methodPropertiesForMethodName.add(methodProperty);
+      } else {
```

Comment:
For readability, consider reversing this conditional, so you have:

```
if (methodName == null || getValueStringFromMap(methodProperty, FLAG_TYPE_KEY) == null) {
   throw ...;
}
if (!isArgumentIndexOptional && getArgumentIndexFromMap(methodProperty) == null) {
   throw ...;
}
List<Map<String, Object>> methodPropertiesForMethodName = ...;
...
```

Reply:
Done.
Also replaced the add methodProperty/new ArrayList logic with a one-liner using `computeIfAbsent`